code/modules/projectiles/ammo_datums.dm

```diff
@@ -2738,6 +2738,67 @@
 	shell_speed = AMMO_SPEED_TIER_2
 	added_spit_delay = 0
 
+
+/datum/ammo/xeno/acid/marking
+	name = "marking spit"
+
+	damage = 15
+	shell_speed = AMMO_SPEED_TIER_2
+	accuracy = HIT_ACCURACY_TIER_10*8
+	flags_ammo_behavior = AMMO_SKIPS_ALIENS| AMMO_XENO
+	damage_falloff = 0
+	max_range = 16
+
+/datum/ammo/xeno/acid/marking/on_hit_mob(mob/target, obj/item/projectile/impacting_projectile)
+	. = ..()
+	target.AddComponent(/datum/component/bonus_damage_stack, 100, 1 SECONDS) //subject to change
+	var/healed_xeno = 0
+	for(var/mob/living/carbon/xenomorph/boosting_xeno in range(5, target))
+		var/hivenumber = XENO_HIVE_NORMAL
+		if (!boosting_xeno.check_state())
+			break
+
+		if (boosting_xeno.hivenumber != hivenumber || boosting_xeno.stat == DEAD)
+			continue
+
+		if (SEND_SIGNAL(boosting_xeno, COMSIG_XENO_PRE_HEAL) & COMPONENT_CANCEL_XENO_HEAL)
+			continue
+
+		to_chat(boosting_xeno, SPAN_XENOBOLDNOTICE("Your carapace regenerates as the acid hits [target]!"))
+		boosting_xeno.gain_health(35)
+		boosting_xeno.flick_heal_overlay(3 SECONDS, "#D9F500")
+		healed_xeno += 1
```

Comment:
healed_xeno++

Reply:
Tried debugging with healed_xeno++, it didn't work